vessim/sil.py

```diff
@@ -143,28 +118,27 @@ def start(self, microgrid: Microgrid) -> None:
 
     def step(self, time: datetime, p_delta: float, actor_infos: dict) -> None:
         assert self.microgrid is not None
-        self.broker._add_microgrid_data(
-            time,
-            {
-                "microgrid": self.microgrid,
-                "actor_infos": actor_infos,
-                "p_delta": p_delta,
-            },
+        self.data_pipe_in.send(
+            (
+                time,
+                {
+                    "microgrid": self.microgrid,
+                    "actor_infos": actor_infos,
+                    "p_delta": p_delta,
+                },
+            )
         )
 
-    def finalize(self) -> None:
-        self.broker._finalize()
-
     def _collect_set_requests_loop(self):
         while True:
             events_by_category = defaultdict(dict)
-            for event in self.broker._consume_events():
+            while self.events_pipe_out.poll():
+                event = self.events_pipe_out.recv()
                 events_by_category[event["category"]][event["time"]] = event["value"]
-            for category, events in events_by_category.items():
+            for category, _ in events_by_category.items():
                 self.request_collectors[category](
                     events=events_by_category[category],
                     microgrid=self.microgrid,
-                    kwargs=self.kwargs,
                 )
             sleep(self.request_collector_interval)
```

Comment:
Actually, the `request_collector_interval` does not serve a good purpose anymore, or? The only thing it does now, is to delay when an event is processed - the thread will go to sleep while waiting for the pipe anyway.

But maybe it is useful to delay the processing of events in some cases :)